### PR TITLE
SFR-1969: Add "Read Online" functionality for UP items

### PR DESCRIPTION
**Pull request by jackiequach**
[Jira Ticket](https://newyorkpubliclibrary.atlassian.net/browse/SFR-1969)

### This PR does the following:
- Adds "Read Online" functionality for University Press items by passing custom fetcher to the web reader

### Testing requirements & instructions: 
-

### CODE REVIEW

**Comment by kylevillegas93** [comment]
Nice work! Logic looks good. I'll wrap up my review tomorrow.

File: src/util/EditionCardUtils.tsx [hunk 1]
```diff
@@ -220,4 +221,39 @@ export default class EditionCardUtils {
         : undefined;
     return universityPress !== undefined;
   }
+
+  static createGetContent = (nyplIdentityCookie: any, router: NextRouter) => {
+    const fetchWithAuth = async (fulfillUrl: string, proxyUrl?: string) => {
```
Comment (kylevillegas93) [inline src/util/EditionCardUtils.tsx:226]: Looks like this code could be reused to call other endpoints that need authentication besides the fulfill endpoint?

I think it's fine to keep it like this for now but we could consider abstracting this out to use with other endpoints later on.

**Review by kylevillegas93 — APPROVED** [review]
Approving based on the logic - but will leave the rest to the frontend squad

**Review by oliviawongnyc — APPROVED** [review]
Had a couple small comments, but going ahead and approving.

File: src/components/ReaderLayout/ReaderLayout.tsx [hunk 2]
```diff
@@ -185,6 +194,11 @@ const ReaderLayout: React.FC<{
           pdfWorkerSrc={pdfWorkerSrc}
           headerLeft={<BackButton />}
           injectablesFixed={injectables}
+          getContent={
```
[the remark this answers is not in the document]
Comment (jackiequach) [reply src/components/ReaderLayout/ReaderLayout.tsx:197]: This should be resolved with npm i. The newest version of the web-reader updates this prop definition.

[the remark this answers is not in the document]
Comment (oliviawongnyc) [reply src/components/ReaderLayout/ReaderLayout.tsx:197]: Yup – thank you!

**Pull request merged**